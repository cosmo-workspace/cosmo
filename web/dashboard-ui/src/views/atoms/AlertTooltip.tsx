import {
  darken, lighten, styled, Tooltip, tooltipClasses, TooltipProps
} from "@mui/material";
import React from "react";

export const AlertTooltip = styled(({ className, ...props }: TooltipProps) => (
  <Tooltip {...props} arrow classes={{ popper: className }} />
))(({ theme }) => ({
  [`& .${tooltipClasses.arrow}`]: {
    fontSize: '1rem',
    color: (theme.palette.mode === 'light' ? lighten : darken)(theme.palette['info'].light, 0.9),
  },
  [`& .${tooltipClasses.tooltip}`]: {
    borderRadius: "4px",
    boxShadow: theme.shadows[5],
    backgroundColor: (theme.palette.mode === 'light' ? lighten : darken)(theme.palette['info'].light, 0.9),
  },
}));
