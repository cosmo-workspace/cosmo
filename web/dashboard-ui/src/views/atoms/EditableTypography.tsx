import { Check, Close, Edit } from "@mui/icons-material";
import {
  IconButton,
  InputBase,
  InputBaseProps,
  Stack,
  Tooltip,
} from "@mui/material";
import React, { useState } from "react";

export type EditableTypographyProps = InputBaseProps & {
  children: string;
  onSave: (inputData: string) => void;
  showAlways?: boolean;
};

export const EditableTypography: React.FC<EditableTypographyProps> = ({
  children,
  onSave,
  showAlways,
  ...props
}) => {
  const [showEditIcon, setShowEditIcon] = useState(showAlways);
  const [editting, setEditing] = useState(false);
  const [inputData, setInputData] = useState<string>(children);

  const editIcon = (
    <Tooltip title="Edit" placement="top">
      <IconButton
        disableRipple
        size="small"
        type="button"
        sx={{ ml: 1 }}
        onClick={() => {
          setEditing(true);
        }}
      >
        <Edit />
      </IconButton>
    </Tooltip>
  );
  const editingIcons = (
    <Stack direction="row">
      <Tooltip title="Save" placement="top">
        <IconButton
          disableRipple
          size="small"
          type="button"
          sx={{ ml: 1 }}
          onClick={() => {
            onSave(inputData);
            setEditing(false);
          }}
        >
          <Check />
        </IconButton>
      </Tooltip>
      <Tooltip title="Cancel" placement="top">
        <IconButton
          disableRipple
          size="small"
          type="button"
          sx={{ ml: 1 }}
          onClick={() => {
            setInputData(children);
            setEditing(false);
          }}
        >
          <Close />
        </IconButton>
      </Tooltip>
    </Stack>
  );

  return (
    <InputBase
      size="small"
      fullWidth
      placeholder="Name"
      endAdornment={
        editting ? editingIcons : showEditIcon ? editIcon : undefined
      }
      defaultValue={children}
      readOnly={!editting}
      autoFocus={editting}
      onMouseOver={() => {
        setShowEditIcon(showAlways || true);
      }}
      onMouseLeave={() => {
        setShowEditIcon(showAlways || false);
      }}
      onBlur={(e) => {
        setInputData(e.currentTarget.value);
      }}
      sx={{ borderBottom: editting ? 1 : 0 }}
      {...props}
    />
  );
};
