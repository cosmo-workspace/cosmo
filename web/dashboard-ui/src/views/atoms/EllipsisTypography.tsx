import {
  Tooltip,
  TooltipProps,
  Typography,
  TypographyProps,
} from "@mui/material";
import React from "react";

export type EllipsisTypographyProps = Omit<TypographyProps, "children"> & {
  children?: string;
  placement?: TooltipProps["placement"];
};

export const EllipsisTypography: React.FC<EllipsisTypographyProps> = (
  props
) => {
  const [isOverflow, setIsOverflow] = React.useState(false);
  const paragraph = React.useRef<HTMLSpanElement>(null);

  React.useEffect(() => {
    const pElement = paragraph.current;
    if (pElement) {
      setIsOverflow(Boolean(pElement.offsetWidth < pElement.scrollWidth));
    }
  }, [paragraph]);

  const title = props.children || "";

  // return isOverflow ? (<Tooltip title={title} placement={placement} > {typoglaphy} </Tooltip >) : typoglaphy
  return isOverflow ? (
    <Tooltip title={title} placement={props.placement}>
      <Typography
        ref={paragraph}
        variant="caption"
        display="block"
        sx={{
          textOverflow: "ellipsis",
          whiteSpace: "nowrap",
          overflow: "hidden",
        }}
        {...props}
      >
        {props.children}
      </Typography>
    </Tooltip>
  ) : (
    <Typography
      ref={paragraph}
      variant="caption"
      display="block"
      sx={{
        textOverflow: "ellipsis",
        whiteSpace: "nowrap",
        overflow: "hidden",
      }}
      {...props}
    >
      {props.children}
    </Typography>
  );
};
