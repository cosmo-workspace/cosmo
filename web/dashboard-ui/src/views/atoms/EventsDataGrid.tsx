import { Info, Warning } from "@mui/icons-material";
import { Chip, SxProps } from "@mui/material";
import {
  DataGrid,
  DataGridProps,
  GridColDef,
  gridClasses,
} from "@mui/x-data-grid";
import React from "react";
import { Event } from "../../proto/gen/dashboard/v1alpha1/event_pb";
import { EventDetailDialogContext } from "../organisms/EventDetailDialog";

export type EventsDataGridProp = {
  events: Event[];
  maxHeight?: number;
  sx?: SxProps;
  dataGridProps?: Omit<DataGridProps, "columns" | "sx">;
  clock: Date;
};

export const EventsDataGrid: React.FC<EventsDataGridProp> = ({
  events,
  maxHeight,
  sx,
  dataGridProps,
  clock,
}) => {
  const eventDetailDialogDispatch = EventDetailDialogContext.useDispatch();

  const columns: GridColDef[] = [
    { field: "type", headerName: "Type", width: 50 },
    {
      field: "eventTime",
      headerName: "LastSeen",
      valueGetter: (value, row: Event) =>
        (row.series?.lastObservedTime || row.eventTime)?.toDate(),
      valueFormatter: (value: Date) =>
        formatTime(clock.getTime() - value.getTime()),
      flex: 0.2,
      minWidth: 80,
    },
    {
      field: "reason",
      headerName: "Reason",
      flex: 0.25,
      renderCell: (params) => (
        <Chip
          icon={
            params.api.getRow(params.id).type == "Normal" ? (
              <Info color="success" />
            ) : (
              <Warning color="warning" />
            )
          }
          label={params.value}
        />
      ),
    },
    { field: "regardingWorkspace", headerName: "Workspace" },
    {
      field: "regarding",
      headerName: "Object",
      valueGetter: (value, row) =>
        row.regarding?.kind + "/" + row.regarding?.name,
      flex: 0.3,
    },
    { field: "reportingController", headerName: "Reporter" },
    {
      field: "series",
      headerName: "Count",
      valueGetter: (value, row) => row.series?.count || 1,
    },
    { field: "note", headerName: "Message", flex: 1 },
  ];

  return (
    <>
      <div style={{ width: "100%", maxHeight: maxHeight, minHeight: 100 }}>
        <DataGrid
          autoHeight={maxHeight === undefined}
          rows={events}
          columns={columns}
          getRowHeight={() => "auto"}
          sx={{
            [`& .${gridClasses.cell}`]: {
              py: 1,
            },
            ...sx,
          }}
          onRowDoubleClick={(params) => {
            eventDetailDialogDispatch(true, { event: params.row });
          }}
          hideFooter
          {...dataGridProps}
        />
      </div>
    </>
  );
};

export function formatTime(milisec: number): string {
  const sec = Math.floor(milisec / 1000) % 60;
  const min = Math.floor(milisec / 1000 / 60) % 60;
  const hours = Math.floor(milisec / 1000 / 60 / 60) % 24;
  if (hours > 0) return `${hours}h${min}m${sec}s`;
  if (min > 0) return `${min}m${sec}s`;
  return `${sec}s`;
}
