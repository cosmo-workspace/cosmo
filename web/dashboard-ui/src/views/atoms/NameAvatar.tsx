import { Avatar, AvatarProps, Typography } from "@mui/material";
import { AccountCircle } from "@mui/icons-material";
import React from "react";

export const NameAvatar: React.VFC<{ name?: string } & AvatarProps> = (props) => {
  return (
    props.name ?
      <Avatar {...props} sx={{ bgcolor: stringToColor(props.name), ...props.sx }}>
        <Typography sx={{ color: (theme) => (theme.palette.mode === 'light' ? 'white' : 'black') }} fontSize='inherit'>
          {props.name.substring(0, 1).toUpperCase()}
        </Typography>
      </Avatar>
      :
      <Avatar {...props} ><AccountCircle /></Avatar>
  )
}

const stringToColor = (str: string) => {
  let hash = 0;
  for (let i = 0; i < str.length; i++) {
    hash = str.charCodeAt(i) + ((hash << 5) - hash);
  }
  let color = '#';
  for (let i = 0; i < 3; i++) {
    let value = (hash >> (i * 8)) & 0xFF;
    color += ('00' + value.toString(16)).substr(-2);
  }
  return color;
}
