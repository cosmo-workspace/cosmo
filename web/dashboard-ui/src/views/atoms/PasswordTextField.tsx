import {
  IconButton,
  InputAdornment, TextField, TextFieldProps
} from "@mui/material";
import { Visibility, VisibilityOff, VpnKey } from "@mui/icons-material";
import React, { useState } from "react";


export const PasswordTextField: React.VFC<TextFieldProps> = ({ type, ...props }) => {
  const [isPassShow, setIsPassShow] = useState(false);

  return (
    <TextField {...props} type={isPassShow ? 'text' : 'password'}
      InputProps={{
        startAdornment: (<InputAdornment position="start"><VpnKey /></InputAdornment>),
        endAdornment: (<InputAdornment position="end">
          <IconButton
            tabIndex={-1}
            onPointerDown={() => { setIsPassShow(true) }}
            onPointerUp={() => { setIsPassShow(false) }}
            onPointerOut={() => { setIsPassShow(false) }}
          >
            {isPassShow ? <Visibility /> : <VisibilityOff />}
          </IconButton>
        </InputAdornment>),
      }}
    />
  );
}
