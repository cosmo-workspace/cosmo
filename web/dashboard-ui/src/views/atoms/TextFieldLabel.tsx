import {
  InputAdornment, TextField, TextFieldProps
} from "@mui/material";
import React, { ReactNode } from "react";

export type TextFieldLabelProps =
  Omit<TextFieldProps, 'focused|valiant'>
  & { startAdornmentIcon?: ReactNode };

export const TextFieldLabel: React.VFC<TextFieldLabelProps> = ({ InputProps, startAdornmentIcon, ...props }) => {

  return (
    <TextField {...props} focused={false} variant="filled"
      InputProps={{
        ...InputProps,
        readOnly: true,
        inputProps: { tabIndex: -1 },
        startAdornment: startAdornmentIcon && (<InputAdornment position="start">{startAdornmentIcon}</InputAdornment>),
      }}
    />)
}
