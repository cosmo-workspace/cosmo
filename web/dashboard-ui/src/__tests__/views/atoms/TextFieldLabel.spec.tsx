import { PersonOutlineTwoTone } from '@mui/icons-material';
import { cleanup, render } from '@testing-library/react';
import React from 'react';
import { afterEach, describe, expect, it } from "vitest";
import { TextFieldLabel } from '../../../views/atoms/TextFieldLabel';

afterEach(cleanup);


describe('TextFieldLabel', () => {

  it('TextFieldLabel', () => {
    const { asFragment } = render(
      <TextFieldLabel label='label1' fullWidth value='value1' />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('TextFieldLabel startAdornmentIcon', () => {
    const { asFragment } = render(
      <TextFieldLabel label='label1' fullWidth value='value1' startAdornmentIcon={<PersonOutlineTwoTone />} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

});