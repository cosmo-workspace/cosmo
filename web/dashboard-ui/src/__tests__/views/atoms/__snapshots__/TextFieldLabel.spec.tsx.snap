// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextFieldLabel > TextFieldLabel 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
}

.emotion-1 {
  color: rgba(0, 0, 0, 0.6);
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  padding: 0;
  position: relative;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: calc(133% - 24px);
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: translate(12px, 7px) scale(0.75);
  -moz-transform: translate(12px, 7px) scale(0.75);
  -ms-transform: translate(12px, 7px) scale(0.75);
  transform: translate(12px, 7px) scale(0.75);
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  z-index: 1;
  pointer-events: auto;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-1.Mui-focused {
  color: #1976d2;
}

.emotion-1.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-1.Mui-error {
  color: #d32f2f;
}

.emotion-2 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  position: relative;
  background-color: rgba(0, 0, 0, 0.06);
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  -webkit-transition: background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.emotion-2.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-2:hover {
  background-color: rgba(0, 0, 0, 0.09);
}

@media (hover: none) {
  .emotion-2:hover {
    background-color: rgba(0, 0, 0, 0.06);
  }
}

.emotion-2.Mui-focused {
  background-color: rgba(0, 0, 0, 0.06);
}

.emotion-2.Mui-disabled {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-2:after {
  border-bottom: 2px solid #1976d2;
  left: 0;
  bottom: 0;
  content: "";
  position: absolute;
  right: 0;
  -webkit-transform: scaleX(0);
  -moz-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
  -webkit-transition: -webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  pointer-events: none;
}

.emotion-2.Mui-focused:after {
  -webkit-transform: scaleX(1) translateX(0);
  -moz-transform: scaleX(1) translateX(0);
  -ms-transform: scaleX(1) translateX(0);
  transform: scaleX(1) translateX(0);
}

.emotion-2.Mui-error:before,
.emotion-2.Mui-error:after {
  border-bottom-color: #d32f2f;
}

.emotion-2:before {
  border-bottom: 1px solid rgba(0, 0, 0, 0.42);
  left: 0;
  bottom: 0;
  content: "\\00a0";
  position: absolute;
  right: 0;
  -webkit-transition: border-bottom-color 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: border-bottom-color 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  pointer-events: none;
}

.emotion-2:hover:not(.Mui-disabled, .Mui-error):before {
  border-bottom: 1px solid rgba(0, 0, 0, 0.87);
}

.emotion-2.Mui-disabled:before {
  border-bottom-style: dotted;
}

.emotion-3 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding-top: 25px;
  padding-right: 12px;
  padding-bottom: 8px;
  padding-left: 12px;
}

.emotion-3::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3:focus {
  outline: 0;
}

.emotion-3:invalid {
  box-shadow: none;
}

.emotion-3::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-3.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-3:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-3:-webkit-autofill {
  border-top-left-radius: inherit;
  border-top-right-radius: inherit;
}

<div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root emotion-0"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled emotion-1"
      data-shrink="true"
      for=":r0:"
      id=":r0:-label"
    >
      label1
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl Mui-readOnly MuiInputBase-readOnly emotion-2"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiFilledInput-input Mui-readOnly MuiInputBase-readOnly emotion-3"
        id=":r0:"
        readonly=""
        tabindex="-1"
        type="text"
        value="value1"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextFieldLabel > TextFieldLabel startAdornmentIcon 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
}

.emotion-1 {
  color: rgba(0, 0, 0, 0.6);
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  padding: 0;
  position: relative;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: calc(133% - 24px);
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: translate(12px, 7px) scale(0.75);
  -moz-transform: translate(12px, 7px) scale(0.75);
  -ms-transform: translate(12px, 7px) scale(0.75);
  transform: translate(12px, 7px) scale(0.75);
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  z-index: 1;
  pointer-events: auto;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-1.Mui-focused {
  color: #1976d2;
}

.emotion-1.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-1.Mui-error {
  color: #d32f2f;
}

.emotion-2 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  position: relative;
  background-color: rgba(0, 0, 0, 0.06);
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  -webkit-transition: background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  padding-left: 12px;
}

.emotion-2.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-2:hover {
  background-color: rgba(0, 0, 0, 0.09);
}

@media (hover: none) {
  .emotion-2:hover {
    background-color: rgba(0, 0, 0, 0.06);
  }
}

.emotion-2.Mui-focused {
  background-color: rgba(0, 0, 0, 0.06);
}

.emotion-2.Mui-disabled {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-2:after {
  border-bottom: 2px solid #1976d2;
  left: 0;
  bottom: 0;
  content: "";
  position: absolute;
  right: 0;
  -webkit-transform: scaleX(0);
  -moz-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
  -webkit-transition: -webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  pointer-events: none;
}

.emotion-2.Mui-focused:after {
  -webkit-transform: scaleX(1) translateX(0);
  -moz-transform: scaleX(1) translateX(0);
  -ms-transform: scaleX(1) translateX(0);
  transform: scaleX(1) translateX(0);
}

.emotion-2.Mui-error:before,
.emotion-2.Mui-error:after {
  border-bottom-color: #d32f2f;
}

.emotion-2:before {
  border-bottom: 1px solid rgba(0, 0, 0, 0.42);
  left: 0;
  bottom: 0;
  content: "\\00a0";
  position: absolute;
  right: 0;
  -webkit-transition: border-bottom-color 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: border-bottom-color 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  pointer-events: none;
}

.emotion-2:hover:not(.Mui-disabled, .Mui-error):before {
  border-bottom: 1px solid rgba(0, 0, 0, 0.87);
}

.emotion-2.Mui-disabled:before {
  border-bottom-style: dotted;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 0.01em;
  max-height: 2em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  color: rgba(0, 0, 0, 0.54);
  margin-right: 8px;
}

.emotion-3.MuiInputAdornment-positionStart.emotion-3:not(.MuiInputAdornment-hiddenLabel) {
  margin-top: 16px;
}

.emotion-4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-5 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding-top: 25px;
  padding-right: 12px;
  padding-bottom: 8px;
  padding-left: 0;
}

.emotion-5::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-5::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-5:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-5::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-5:focus {
  outline: 0;
}

.emotion-5:invalid {
  box-shadow: none;
}

.emotion-5::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-5.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-5:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-5:-webkit-autofill {
  border-top-left-radius: inherit;
  border-top-right-radius: inherit;
}

<div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root emotion-0"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled emotion-1"
      data-shrink="true"
      for=":r1:"
      id=":r1:-label"
    >
      label1
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart Mui-readOnly MuiInputBase-readOnly emotion-2"
    >
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-filled MuiInputAdornment-sizeMedium emotion-3"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-4"
          data-testid="PersonOutlineTwoToneIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="8"
            opacity=".3"
            r="2.1"
          />
          <path
            d="M12 14.9c-2.97 0-6.1 1.46-6.1 2.1v1.1h12.2V17c0-.64-3.13-2.1-6.1-2.1z"
            opacity=".3"
          />
          <path
            d="M12 13c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4zm6.1 5.1H5.9V17c0-.64 3.13-2.1 6.1-2.1s6.1 1.46 6.1 2.1v1.1zM12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0-6.1c1.16 0 2.1.94 2.1 2.1 0 1.16-.94 2.1-2.1 2.1S9.9 9.16 9.9 8c0-1.16.94-2.1 2.1-2.1z"
          />
        </svg>
      </div>
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedStart Mui-readOnly MuiInputBase-readOnly emotion-5"
        id=":r1:"
        readonly=""
        tabindex="-1"
        type="text"
        value="value1"
      />
    </div>
  </div>
</DocumentFragment>
`;
